src/tiered/tiered_opener.c

```diff
@@ -0,0 +1,75 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __tiered_opener_open --
+ *     Open an object by number.
+ */
+int
+__tiered_opener_open(WT_BLOCK_FILE_OPENER *opener, WT_SESSION_IMPL *session, void *cookie,
+  uint64_t object_id, WT_FS_OPEN_FILE_TYPE type, u_int flags, WT_FH **fhp)
+{
+    WT_BUCKET_STORAGE *bstorage;
+    WT_DECL_RET;
+    WT_TIERED *tiered;
+    const char *object_name;
+
+    WT_UNUSED(opener);
+    tiered = (WT_TIERED *)cookie;
+
```

Comment:
I don't know if you looked at it but the function that exists that you might want to use (maybe with some small changes) is `tiered/tiered_handle.c:__wt_tiered_name`.

Reply:
Done.